src/test.rs

```diff
@@ -140,10 +141,30 @@ mod marlin {
 
             test_canonical_serialize_deserialize(true, &proof);
 
-            assert!(Marlin::<F, PC, D>::verify(&index_vk, &pc_vk,&[c, d], &proof).unwrap());
+            assert!(Marlin::<F, PC, D>::verify(&index_vk, &pc_vk, &[c, d], &proof).unwrap());
             println!("Called verifier");
             println!("\nShould not verify (i.e. verifier messages should print below):");
             assert!(!Marlin::<F, PC, D>::verify(&index_vk, &pc_vk, &[a, a], &proof).unwrap());
+
+            let circ_fake = Circuit {
```

Comment:
Would've been nice to change the previous proof and modify it so that it fails in the verify opening proof, rather than generating new parameters and new proof. Is it possible ?

Reply:
I've changed the implementation and added the randomization for proof